impl/src/main/java/org/ehcache/UserManagedCacheBuilder.java

```diff
@@ -241,6 +259,11 @@ public void close() throws Exception {
         storeProvider.releaseStore(store);
       }
     });
+
+    if (this.eventNotifier instanceof DisabledCacheEventNotificationService && (orderedExecutor != null & unOrderedExecutor != null)) {
+      this.eventNotifier = new CacheEventDispatcherImpl<K, V>(store, unOrderedExecutor, orderedExecutor);
```

Comment:
So this works indeed. But if the user tries to provide a different implementation of `CacheEventDispatcher`, it will never be connected to the `Store`.
To have a working solution, you need to be able to pass the `CacheEventDispatcherImpl` from the outside and get a working setup. And with the current contract, this does not work, which indicates that we are hiding an implicit contract. We need to make it explicit. That's why I mentioned that the `CacheEventDispatcher` probably needs a setter for the `StoreEventSource`.